### PR TITLE
A javascript-focused template group

### PR DESCRIPTION
**Pull request by zachthompson**
Add a new template group with a javascript file in the configuration.  Also added a semi-modified Perl module template that puts the query remainder directly in the data, i.e. the processing will be done in the js.

See if this makes sense or if we just want to have people roll through the optional templates and select js to do the same thing.

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: template/lib/DDG/Goodie/Minimal.pm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            # Name - Used for Answer Bar Tab
+            # Value should be chosen from existing Instant Answer topics
+            # see https://duck.co/duckduckhack/display_reference#codenamecode-emstringem-required
+            name => 'Answer',
+
+            data => {
+                remainder => $_
+            },
+
+            templates => {
```
Comment (moollaza) [inline template/lib/DDG/Goodie/Minimal.pm]: is this required? might as well define this once in the JS?

File: template/lib/DDG/Goodie/Minimal.pm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+use DDG::Goodie;
+use strict;
+
+zci answer_type => '<: $ia_id :>';
+
+# Caching - https://duck.co/duckduckhack/spice_advanced_backend#caching-api-responses
+zci is_cached   => 1;
+
+# Triggers - https://duck.co/duckduckhack/goodie_triggers
+triggers any => 'triggerWord', 'trigger phrase';
```
Comment (moollaza) [inline template/lib/DDG/Goodie/Minimal.pm]: just realized we have an uppercase letter here :frowning: 

That's my bad -- we should change that to `triggerword`

File: template/lib/DDG/Goodie/Minimal.pm (hunk 1 above)
Comment (moollaza) [inline template/lib/DDG/Goodie/Minimal.pm]: Seems like it's required...

**Comment by moollaza** [comment]
@zachthompson this LGTM. It would be great if the Perl didn't have to pass along a `template` obj so that the JS would handle *everything*.

Otherwise this works great.

**Comment by zachthompson** [comment]
Totally agree about having to define the template here.  Had the same thought.  Unfortunately that's how we currently determine whether it's templated or not so we can treat it differently.  With autotemplating maybe going away completely, we should revisit and see if we can change this in the future.

**Comment by moollaza** [comment]
:+1:

**Pull request merged**